File: c/basic/binary_to_decimal.c
/*
 * Compile With:
 *   gcc -Wall -o binary_to_decimal binary_to_decimal.c
 * Run: ./binary_to_decimal
 */
#include <stdio.h>
const char QUIT = 'q';

int main(){

  int originalNumber, num;
  int base = 1;
  int decimalNumber = 0;
  char input[50];
  char quit;

  while(1){
    printf("Enter a number in binary form (1's and 0's):");
    fgets(input, 50, stdin);

    sscanf(input, "%c", &quit);
    if (quit == QUIT){
      printf("Goodbye\n");
      return 0;
    } else {
      sscanf(input, "%d", &originalNumber);
      num = originalNumber;
      while (num > 0){
        int rem = num % 10;
        decimalNumber = decimalNumber + rem * base;
        num = num / 10;
        base = base * 2;
      }
      printf("Your number in decimal form is: %d\n", decimalNumber);
    }
    decimalNumber = 0;
    base = 1;
  }
  return 0;
}

File: ruby/basic/binary_to_decimal.rb
loop do
  puts "Enter a number in binary form"

  input = gets.chomp
  if input == 'q'
    puts "Goodbye"
    break;
  else
    original_num = input.to_i
    decimal_num = 0
    base = 1
    while original_num > 0
      rem = original_num % 10
      decimal_num = decimal_num + rem * base
      base = base * 2
      original_num = original_num / 10
    end
    puts "Your number in decimal form is #{decimal_num}"
  end
end
